src/main/java/nextstep/courses/domain/PaidSession.java

```diff
@@ -11,17 +11,21 @@ public class PaidSession extends Session implements SessionStrategy {
 
     private int studentMaxCount;
 
-    public PaidSession(SessionInfo sessionInfo, SessionImage sessionImage,
-                       long salePrice, StateCode stateCode, int studentMaxCount) {
-        super(sessionInfo, sessionImage, salePrice, stateCode, SessionType.PAID);
+    public PaidSession(SessionInfo sessionInfo, SessionImages sessionImages, SessionPrice sessionPrice, int studentMaxCount) {
+        super(sessionInfo, sessionImages, sessionPrice, SessionType.PAID);
         this.studentMaxCount = studentMaxCount;
+
+        if(sessionPrice.getSalePrice() <= 0 ){
```

Comment:
```suggestion
        if (sessionPrice.isLessThanOrEqualTo(0) ){
```
값 객체를 활용하면 더 좋을 것 같네요 😄